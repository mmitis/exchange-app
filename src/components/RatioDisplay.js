//@flow
import React from "react";

type Props = {
    ratio: number,
    ratioCharFrom: string,
    ratioCharTo: string
}
export const RatioDisplay = ({
                                  ratio, ratioCharFrom, ratioCharTo
} : Props) => {
    return (
            <div className="ratio-display">
                <span className="ratio-display__counter">{ratioCharFrom}1 = {ratioCharTo}{ratio.toFixed(2)}</span>
            </div>
    )
};

export default React.memo(RatioDisplay);
