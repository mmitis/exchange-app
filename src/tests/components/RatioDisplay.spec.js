import RatioDisplay from './../../components/RatioDisplay';
import { mount } from 'enzyme';
import React from "react";

describe('Component: RatioDisplay', () => {

    it('test simple render', async () => {
        const props = {
            ratio:3,
            ratioCharFrom: 'PLN',
            ratioCharTo: 'EUR'
        };
        const ratio = mount(<RatioDisplay
            {...props}
        />);
        
        expect(ratio).toMatchSnapshot();
    });

});
