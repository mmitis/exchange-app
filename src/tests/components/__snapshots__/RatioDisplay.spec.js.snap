// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: RatioDisplay test simple render 1`] = `ReactWrapper {}`;
